puffin_egui/src/flamegraph.rs

```diff
@@ -731,19 +757,17 @@ fn paint_scope(
         }
 
         if result == PaintResult::Hovered {
+            let Some(scope_details) = info.scope_collection.fetch_by_id(&scope.id) else {
+                return Ok(PaintResult::Culled);
+            };
             egui::show_tooltip_at_pointer(&info.ctx, Id::new("puffin_profiler_tooltip"), |ui| {
-                ui.monospace(format!("id:       {}", scope.record.id));
-                if !scope.record.location.is_empty() {
-                    ui.monospace(format!("location: {}", scope.record.location));
-                }
-                if !scope.record.data.is_empty() {
-                    ui.monospace(format!("data:     {}", scope.record.data));
-                }
+                paint_scope_details(ui, scope.id, scope.record.data, scope_details);
+                add_space(ui);
                 ui.monospace(format!(
                     "duration: {:7.3} ms",
                     to_ms(scope.record.duration_ns)
                 ));
-                ui.monospace(format!("children: {num_children}"));
+                ui.monospace(format!("children: {num_children:3.3}"));
```

Comment:
three decimals seems overkill for an integer ;)

```suggestion
                ui.monospace(format!("children: {num_children:3}"));
```